### ISSUE TITLE
Pytorch Point to Plane similar to SimpleICP

### ISSUE THREAD

**Issue by LemonPi** [pglira/simpleICP#27]
Hi, I tried out SimpleICP and it seems to work very well for my problem. The only issue with SimpleICP for me is that it's very slow due to not being batched (as I need a distribution of ICP results). I was wondering if you knew any pytorch implementations of point to plane ICP? pytorch3d's implementation I believe is point to point and produces much worse fits than SimpleICP. If not, I am interested in porting this python implementation to pytorch to support batching.

**Comment by pglira** [comment]
Hi, its not completely clear to me what you mean with "batching". Can you clarify this? Is the goal to run register n point cloud pairs at the same time, i.e. in parallel?

**Comment by LemonPi** [comment]
The goal is to register two point clouds together but with different initial pose estimates in parallel. This is achieved in pytorch3d by passing the point cloud as `B x N x 3` and `B x M x 3` inputs with `B x 4 x 4` initial homogeneous transform estimates. (This formulation allows registration of different pairs of point clouds in parallel, but I just use it to duplicate my input point clouds B times).

**Comment by pglira** [comment]
Thanks for the suggestion, but this is out of scope. simpleICP is meant to be a simple, readable reference implementation. A PyTorch port with batching would be a separate project with very different goals.

Closing.
